### DISCUSSION TITLE
Query parameters with hyphens

### DISCUSSION THREAD

**Discussion by yiannis-kt** [fastapi/fastapi#7926] [Questions] [answered]
**Description**

How can I get a query parameter that contains hyphens?

**Comment by Bobronium** [comment, marked as answer]
You can use [aliases](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#alias-parameters) 
Also, I found this for headers: [Header Params automatic conversion](https://fastapi.tiangolo.com/tutorial/header-params/#automatic-conversion).

I wish there was a simpler way to auto-convert all params, like it can be done with  [Config.alias_generator](https://pydantic-docs.helpmanual.io/usage/model_config/#alias-generator) in pydantic

**Comment by yiannis-kt** [comment]
Many thanks. Yes show that for header with option to disable the auto-conversion. Somehow missed that for query params.

**Comment by tiangolo** [comment]
Thanks for the help here @MrMrRobat ! :clap: :bow: 

Thanks for reporting back and closing the issue @yiannis-kt  :+1:
